woocommerce/woocommerce-android
# [Product discount] Final touches

### Product discount - final PR

Closes: #9352 


### Description

This PR adds the final pieces of the "Product discount" feature in order to make it ready for release. List of changes includes:
1. Enabled feature flag
2. Added analytic events - 
3. Added "Total Discounts" row in order create/edit screen
4. Added "Item Discount" row to product list in order create/edit screen
5. Improved navigation flow - clicking "Done" button in discount add/edit screen  redirects back directly to order create/edit screen. This improves the UX and fixes errors when order discount was not applied.
6. Disabled "+ Add discount" and "Edit" buttons in product details screen in case there are active coupons applied. This is caused by the backend issue rejecting discounts in such case. More context here: p1689084213840619/1689083832.564069-slack-C025A8VV728
7. Updated release notes

### Testing instructions

* Verify that after entering discount amount and clicking "Done" we're redirected to the order create/edit screen and discount is applied to the order
* Verify the same when clicking "Remove discount" button
* Verify new discount rows in order create/edit screen as described above and shown below
* Verify analytics events are tracked in logcat

### Images/gif
#### Order create/edit - new Item discount and Discount Total rows

<img src=https://github.com/woocommerce/woocommerce-android/assets/4527432/bb69754d-8a33-4454-990a-e4191386f9ff width=320/>

#### Track events
<img width="1123" alt="Screenshot 2023-07-13 at 16 54 42" src="https://github.com/woocommerce/woocommerce-android/assets/4527432/d4c81a2e-c4f5-4ab7-8fba-dda7efd107a2">
